### PR TITLE
Campsite update modules

### PR DESCRIPTION
**Pull request by ksdenisova**
**Campsite PR**

- Separate modules for the HttpServer and HttpServerFixture
- Separate `Routes` from the HttpServer and HttpServerFixture

Co-authored-by: @Aoxian

### CODE REVIEW

File: lib/http_server/http_server.ex [hunk 1]
```diff
@@ -7,7 +9,7 @@ defmodule HttpServer do
 
   defp loop_acceptor(socket) do
     {:ok, client} = :gen_tcp.accept(socket)
-    {:ok, pid} = Task.Supervisor.start_child(HttpServer.TaskSupervisor, fn -> serve(client) end)
+    {:ok, pid} = Task.Supervisor.start_child(HttpServer.HttpServer.TaskSupervisor, fn -> serve(client) end)
```
Comment (Aoxian) [inline lib/http_server/http_server.ex:12]: maybe alias this?
```elixir
alias HttpServer.HttpServer
```

File: lib/http_server/http_server.ex [hunk 2]
```diff
@@ -16,7 +18,7 @@ defmodule HttpServer do
     response = 
       case read_request(socket) do
         {:ok, http_message} ->
-          http_message |> router.handle_request()
+          http_message |> router.handle_request(Routes)
```
Comment (Aoxian) [inline lib/http_server/http_server.ex:21]: I think we can move Routes to the `HttpServer.Router`

File: lib/http_server/router.ex [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+    alias HttpServer.Request.Parser
+    alias HttpServer.Response
+    alias HttpServer.Response.Builder
 
-    def handle_request(http_message, routes \\ Routes) do
+    def handle_request(http_message, routes) do
         request = parse_http_message(http_message)
 
-        handler = route_to_handler(request)
+        handler = route_to_handler(routes, request)
```
Comment (Aoxian) [inline lib/http_server/router.ex:10]: If we move `Routes` out of `HttpServer.HttpServer` 
```suggestion
        handler = route_to_handler(request)
```

File: lib/http_server/router.ex [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
-    alias Response.Builder
-    alias Request.Parser
+defmodule HttpServer.Router do
+    alias HttpServer.Exception
+    alias HttpServer.Request.Parser
+    alias HttpServer.Response
+    alias HttpServer.Response.Builder
 
-    def handle_request(http_message, routes \\ Routes) do
+    def handle_request(http_message, routes) do
```
Comment (Aoxian) [inline lib/http_server/router.ex:7]: If we move `Routes` out of `HttpServer.HttpServer` 
```suggestion
    def handle_request(http_message) do
```

File: lib/http_server/router.ex [hunk 5]
```diff
@@ -1,17 +1,19 @@
...
         
         response = handler.handle(request, Builder)
         Response.format(response)
     end
 
-    def route_to_handler(request, routes \\ Routes) do
+    def route_to_handler(routes, request) do
         routes.build_routes_map()
         |> get_method_routes(request.method) 
         |> get_handler(request.resource)
```
Comment (Aoxian) [inline lib/http_server/router.ex:19]: If we move `Routes` out of `HttpServer.HttpServer` 

```suggestion
    def route_to_handler(request) do
        Routes.build_routes_map()
        |> get_method_routes(request.method) 
        |> get_handler(request.resource)
```

Comment (Aoxian) [reply lib/http_server/router.ex:19]: I actually think you may have suggested this earlier, but I only see it now.

File: lib/http_server_fixture/handler/simple_post.ex [hunk 6]
```diff
@@ -1,7 +1,7 @@
-defmodule SimplePostHandler do
-    @behaviour Handler
+defmodule HttpServerFixture.Handler.SimplePost do
+    @behaviour HttpServer.Handler
```
Comment (Aoxian) [inline lib/http_server_fixture/handler/simple_post.ex:2]: is `@behavior` being used? GitHub Actions is throwing warning.

File: lib/http_server_fixture/handler/simple_post.ex [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
 
-    @impl Handler
+    @impl HttpServer.Handler
```
Comment (Aoxian) [inline lib/http_server_fixture/handler/simple_post.ex:4]: is `@impl HttpServer.Handler` missing a callback? (GitHub Actions is throwing a warning)

**Comment by Aoxian** [comment]
I know I contributed to some of these changes, but I left comments anyway.  Let me know what you think or make changes.

File: lib/http_server_fixture/handler/simple_post.ex (hunk 6 above)
Comment (ksdenisova) [reply lib/http_server_fixture/handler/simple_post.ex:2]: We didn't have warning about `@behavior` and `@impl` before changes. I'll add `HttpServer` before arguments in `HttpServer.Handler` and check if it fixes the problem.

File: lib/http_server/http_server.ex (hunk 1 above)
Comment (ksdenisova) [reply lib/http_server/http_server.ex:12]: Acceptance tests don't pass with this change. Change to this instead:

````
alias HttpServer.HttpServer.TaskSupervisor
````

File: test/http_server/router_test.exs [hunk 8]
```diff
@@ -30,8 +34,8 @@ defmodule RouterTest do
                 headers: [], 
                 body: [""]}
 
-                assert_raise RouteNotFoundError,
-                    fn -> Router.route_to_handler(request, MockRoutes) end
+                assert_raise Exception.RouteNotFoundError,
+                    fn -> Router.route_to_handler(MockRoutes, request) end
```
Comment (ksdenisova) [inline test/http_server/router_test.exs:38]: I started to make the changes you suggested, but `Router` Unit test failed. We won't be able to mock `Routes` for the tests.

File: lib/http_server/router.ex (hunk 5 above)
Comment (ksdenisova) [reply lib/http_server/router.ex:19]: No, I didn't. I think it's a good suggestion, but we we can't mock `Routes` for tests if we make these changes.

Comment (Aoxian) [reply lib/http_server/router.ex:19]: ok, then maybe go back to your original: 
```suggestion
    def route_to_handler(request, routes \\ Routes) do
        routes.build_routes_map()
        |> get_method_routes(request.method) 
        |> get_handler(request.resource)
```
and make sure to:
```elixir
alias Routes
```

Comment (Aoxian) [reply lib/http_server/router.ex:19]: do you think that would work?  all these comments are basically around getting the Routes injection out of the `HttpServer.HttpServer`.  If that doesn't make sense to you, let me know.  I think I approve of this PR either way.

File: test/http_server/router_test.exs (hunk 8 above)
Comment (Aoxian) [reply test/http_server/router_test.exs:38]: I left another comment above about this. I understand the need to keep the injection for mocking, but also it makes sense for this particular function as well.

**Review by Aoxian — APPROVED** [review]

**Pull request merged**